playwright/_impl/_browser_context.py

```diff
@@ -214,8 +215,9 @@ async def add_init_script(
         self, script: str = None, path: Union[str, Path] = None
     ) -> None:
         if path:
-            with open(path, "r") as file:
-                script = file.read()
+            file = await run_in_threadpool(lambda: open(path, mode="r"))
```

Comment:
It looks like you have two repetitive patterns worth extracting: `async read_file` and `async write_file`.